server/src/lfortran-types.ts

```diff
@@ -11,3 +11,34 @@ export interface ExampleSettings {
 export interface ErrorDiagnostics {
   diagnostics: Diagnostic[];
 }
+
+export namespace ASRSymbolType {
+  export const Program: number = 0;
+  export const Module: number = 1;
+  export const Function: number = 2;
+  export const GenericProcedure: number = 3;
+  export const CustomOperator: number = 4;
+  export const ExternalSymbol: number = 5;
+  export const Struct: number = 6;
+  export const Enum: number = 7;
+  export const UnionType: number = 8;
+  export const Variable: number = 9;
+  export const Class: number = 10;
+  export const ClassProcedure: number = 11;
+  export const AssociateBlock: number = 12;
+  export const Block: number = 13;
+  export const Requirement: number = 14;
+  export const Template: number = 15;
+}
+
+export type ASRSymbolType = 0 | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11 | 12 | 13 | 14 | 15;
```

Comment:
I can refactor LFortran to return the LSP types if you prefer. I stuck with the current model of returning the ASR/LFortran types because that is what the diagnostics already do, but I have nothing against the aforementioned.

The LSP enums are of type number instead of string, so if we implemented your idea we would have to call LFortran to get a symbol type's numeric identifier, then call it again to get the name of the identifier, then map the name back to its respective LSP numeric identifier. That's a lot of back-and-forth and regardless what we decide I would prefer to make a single call to LFortran.

Reply:
I've moved the logic for mapping ASR/LFortran types to LSP types from here to my other PR against lfortran: https://github.com/lfortran/lfortran/pull/5585

Microsoft seems to only add to the types (and occasionally deprecate them) as the LSP spec evolves. They do not seem to change definitions otherwise, so using the LSP definitions within LFortran should be long-lived.

Reply:
Yes, the current solution is good. I set the other PR to merge, then you can merge this one.

Yes, the LSP changes infrequently and in a backward compatible manner. The key is to isolate LFortran's internals, so that we are free to change them and things will be working, we don't have to tie LFortran version with the `lfortran-lsp` version.